go/libraries/doltcore/merge/schema_merge_test.go

```diff
@@ -1354,10 +1369,246 @@ var jsonMergeTests = []schemaMergeTest{
+		require.True(t, (bool)(inRange))
+		newDoc, changed, err := document.Replace(ctx, path, jsonVal.(sql.JSONWrapper))
+		require.NoError(t, err)
+		require.True(t, changed)
+		return newDoc
+	}
+
+	delete := func(document sqltypes.MutableJSON, path string) sqltypes.MutableJSON {
+		newDoc, changed, err := document.Remove(ctx, path)
+		require.True(t, changed)
+		require.NoError(t, err)
+		return newDoc
+	}
+
+	var largeJsonMergeTests = []schemaMergeTest{
+		{
+			name:     "json merge",
+			ancestor: *tbl(sch("CREATE TABLE t (id int PRIMARY KEY, a int, b int, j json)")),
+			left:     tbl(sch("CREATE TABLE t (id int PRIMARY KEY, a int, b int, j json)")),
+			right:    tbl(sch("CREATE TABLE t (id int PRIMARY KEY, a int, b int, j json)")),
+			merged:   *tbl(sch("CREATE TABLE t (id int PRIMARY KEY, a int, b int, j json)")),
+			dataTests: []dataTest{
+				{
+					name:     "parallel insertion",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, insert(largeObject, "$.a", 1)),
+					right:    singleRow(1, 1, 2, insert(largeObject, "$.a", 1)),
+					merged:   singleRow(1, 2, 2, insert(largeObject, "$.a", 1)),
+				},
+				{
+					name:     "convergent insertion",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, insert(largeObject, "$.a", 1)),
+					right:    singleRow(1, 1, 2, insert(largeObject, "$.z", 2)),
+					merged:   singleRow(1, 2, 2, insert(insert(largeObject, "$.a", 1), "$.z", 2)),
+				},
+				{
+					name:     "multiple insertions",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, insert(insert(largeObject, "$.a1", 1), "$.z2", 2)),
+					right:    singleRow(1, 1, 2, insert(insert(largeObject, "$.a2", 3), "$.z1", 4)),
+					merged:   singleRow(1, 2, 2, insert(insert(insert(insert(largeObject, "$.z1", 4), "$.z2", 2), "$.a2", 3), "$.a1", 1)),
+				},
+				{
+					name:     "convergent insertion with escaped quotes in keys",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, insert(largeObject, `$."\"a\""`, 1)),
+					right:    singleRow(1, 1, 2, insert(largeObject, `$."\"z\""`, 2)),
+					merged:   singleRow(1, 2, 2, insert(insert(largeObject, `$."\"a\""`, 1), `$."\"z\""`, 2)),
+				},
+				{
+					name:     "parallel modification",
+					ancestor: singleRow(1, 1, 1, largeObject),
+					left:     singleRow(1, 2, 1, set(largeObject, "$.level7", 1)),
+					right:    singleRow(1, 1, 2, set(largeObject, "$.level7", 1)),
+					merged:   singleRow(1, 2, 2, set(largeObject, "$.level7", 1)),
+				},
+				{
+					name:     "convergent modification",
+					ancestor: singleRow(1, 1, 1, insert(largeObject, "$.a", 1)),
+					left:     singleRow(1, 2, 1, set(insert(largeObject, "$.a", 1), "$.level7", 2)),
+					right:    singleRow(1, 1, 2, set(insert(largeObject, "$.a", 1), "$.a", 3)),
+					merged:   singleRow(1, 2, 2, set(insert(largeObject, "$.a", 3), "$.level7", 2)),
+					skip:     true, //doesn't terminate?
```

Comment:
Can you give some more comments explaining what's going on here? Does this get stuck in a loop that never returns or something? If that's the case, I think we need to fix this one before shipping.

Reply:
Thanks for catching. The test works, I just forgot to re-enable it.